WordPress/wordcamp.org
# Blocks: Add a new "Avatar" block to use in Query Loop

The Query Loop block was introduced in WordPress 5.8. This block renders a list of posts (of any type), and gives the editor fine control over the layout and details. The WC post types (ex, speakers) can be rendered in this block, but it misses some of the CPT-specific info; there is no block for the user's avatar.

This PR adds a new block for Avatar, so that wordcamp organizers can create speaker and organizer pages using the Query Loop block, for finer control than the existing Speakers & Organizers blocks. This block is modeled after the featured image block, allowing users to pick from a set of avatar sizes, and supports the duotone feature.

Additionally, I added featured image support to speakers & organizers, and if a featured image is found, it will override the gravatar (fixes #180). This does not put any extra work on organizers unless they want it, since gravatars are still the default (see `WordCamp\Post_Types\Utilities\get_avatar_or_image`).

I plan to add a few other meta-specific blocks, for speaker sessions, session dates, etc, but this is by far the most involved one. Eventually we could also set some patterns that use the query block, which could go into the initial query placeholder.

### Screenshots

A demo of the Speakers in a query block, with my photo replaced by a featured image.

<details>
<summary>Query block code</summary>

```html

<div class="wp-block-query">



</div>

```
</details>

Query block in the editor:

<img width="641" alt="block-query-loop" src="https://user-images.githubusercontent.com/541093/132557571-37dc3a7a-d101-45a1-a327-34dae5bfefc9.png">

Avatar block options (size, link to post):

<img width="691" alt="block-avatar-inspector" src="https://user-images.githubusercontent.com/541093/132557576-5883d794-76a3-4240-afce-374da363df9e.png">

Toggling duotone on/off:

<img width="998" alt="block-duotone" src="https://user-images.githubusercontent.com/541093/132557587-2449b018-254a-4aa8-8d08-f78ab300534c.png">

On the frontend, 128px avatars, duotone to match the TT1 theme:

<img width="609" alt="Screen Shot 2021-09-08 at 1 43 09 PM" src="https://user-images.githubusercontent.com/541093/132558413-27d5d563-e6bb-45f8-9210-708d37a4989a.png">

Frontend with Organizers in the query block, "Voxel Cat" uses my user ID, but shows the featured image instead:

<img width="670" alt="Screen Shot 2021-09-08 at 1 44 34 PM" src="https://user-images.githubusercontent.com/541093/132558593-943f1466-57cf-44f3-b4a8-df984acabbf3.png">

If set, the featured image is also used in the Speakers & Organizers blocks…

<img width="607" alt="Screen Shot 2021-09-08 at 1 45 45 PM" src="https://user-images.githubusercontent.com/541093/132558793-95fbed07-4eba-4b85-a651-a582461d82f2.png">

And on the individual Speaker page – the featured image itself is disabled on the speaker page, to prevent duplicate images.

<img width="1264" alt="Screen Shot 2021-09-08 at 1 46 12 PM" src="https://user-images.githubusercontent.com/541093/132558848-479550fb-b4ce-41f8-99df-16aab25e4d2b.png">

### How to test the changes in this Pull Request:

1. Create some speakers with different gravatar emails, and/or different organizers
2. Optionally set some featured images
3. Add a query block, change the post type to "Speaker" or "Organizer" and click into the template container to edit them
4. Add an Avatar block, play around with the settings
5. It should behave as expected, and show correctly on the frontend

Note: There is a bug in TT1 where center alignment in the editor looks off, but is correct on the frontend; seems to be the case with the featured image block too.
